### PR TITLE
add suxrankselect

### PR DESCRIPTION
**Pull request by jmeintrup**
1. add sux library as git submodule
2. add suxrankselect tests
3. use suxrankselect for subgraph stack

### CODE REVIEW

**Review by andrej-sajenko — DISMISSED** [review]
Hi @jmeintrup,

1) is the build section in the main markdown description file still up to date after these changes?
2) since we are using submodules now, do we have to add the ``git pull --recurse-submodules`` command in .travis file to make the build work?

**Comment by jmeintrup** [comment]
> 
> 
> Hi @jmeintrup,
> 
>     1. is the build section in the main markdown description file still up to date after these changes?
> 
>     2. since we are using submodules now, do we have to add the `git pull --recurse-submodules` command in .travis file to make the build work?

1. yes
2. there are no submodules used: i implemented it directly into our library

**Comment by jmeintrup** [comment]
@andrej-sajenko there is an issue with travis-ci not reporting the correct status

i found this fix: https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154

afaik only the project owner can do these changes, i.e., you.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "sux"]
```
Comment (andrej-sajenko) [inline .gitmodules]: If we do not use submodules, do we still need the .gitmodules file?

**Comment by andrej-sajenko** [comment]
> @andrej-sajenko there is an issue with travis-ci not reporting the correct status
> 
> i found this fix: https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154
> 
> afaik only the project owner can do these changes, i.e., you.

Tried to change it, but I can only select the deprecated check that we are currently using.
Do not know why it is not possible.

File: .gitmodules (hunk 1 above)
Comment (jmeintrup) [reply .gitmodules]: Removed .gitmodules file

**Review by andrej-sajenko — APPROVED** [review]
@jmeintrup Can you update the *Using the Library* section in the README.MD file.

**Pull request merged**